### ISSUE TITLE
Allow more customizability (lima init, various configs, etc.)

### ISSUE THREAD

**Issue by Kitanotori** [rancher-sandbox/rancher-desktop#2708]
### Problem Description

Currently RD provides limited ways for customizing various underlying components. Particularly, it is not obvious how to provide custom lima VM e.g. in case one wants to use Ubuntu-based VM instead of the default Alpine-based one. Also, there's no obvious way for providing custom configurations for some components, e.g. containerd.

### Proposed Solution

Make it easier to switch between underlying lima vm images. It seems override.yaml is currently the most powerful way for overriding defaults, but it is not obvious what would be required in case one wanted to e.g. switch to different VM image (from Alpine-based to Ubuntu-based, etc.). 

Also, the init flow should be better documented. Currently there doesn't seem to be obvious way for e.g. overriding containerd configuration, because it's apparently overwritten every restart by some undocumented script.

**Comment by mook-as** [comment]
Hmm, if you want that degree of customization, wouldn't running Lima directly be a better fit for your use cases? The fact that it's running Lima under the hood is more of an implementation detail (and isn't true on Windows), so if you want to peel back the covers that much, using the underlying project directly sounds like it would be much better long term; after all, your use case is unlikely to be top-of-mind as we do further development.

`override.yaml` is more of an escape hatch, and not a major feature; at some point (and I'd argue that trying to change the underlying image is way past that point) it's much more maintainable for you to skip that layer.

**Comment by Kitanotori** [comment]
@mook-as VM customization is kind of nice-to-have feature, but I feel there should at least be a way to override containerd configuration in case one wants to use different default snapshotter (e.g. stargz), provide registry mirror settings, etc.

**Comment by Kitanotori** [comment]
In case VM customization is not priority, it would be nice if the default image was based on Ubuntu and systemd with cgroup v2 enabled (reason: to match with most prod env setups). systemd is practically the default in all major distros except Alpine (https://en.wikipedia.org/wiki/Systemd#Adoption).

**Comment by aitorpazos** [comment]
@mook-as from my experience I got the needed levers from `override.yaml` (ie: install some host packages needed for longhorn). If supporting multiple base images broaden the scope too much, I am happy with current approach :smile: . Just my 2 cents on this.

**Comment by jlunaq** [comment]
Is it possible to use an ubuntu image by configuring the `override.yaml` file?

I added this configuration and restarted the Rancher Desktop application, however seems to be not working
```yaml
images:
- location: "https://cloud-images.ubuntu.com/releases/22.04/release/ubuntu-22.04-server-cloudimg-amd64.img"
  arch: "x86_64"
```

I'm considering to use colima which seems to be more flexible, but first I want to confirm Rancher Desktop does offer such customization

**Comment by jandubois** [comment]
> Is it possible to use an ubuntu image by configuring the `override.yaml` file?

@jlunaq No, Rancher Desktop relies on the specific configuration of the base image; you cannot change it without breaking Rancher Desktop.
